apps/web/src/components/Filters/index.tsx

```diff
@@ -3,43 +3,69 @@ import type { FC } from "react";
 import { useRouter } from "next/router";
 import type { DateValueType } from "react-tailwindcss-datepicker";
 
+import { validateRange } from "~/utils/validators";
 import { Button } from "~/components/Button";
 import { useBreakpoint } from "~/hooks/useBreakpoint";
 import type { Option } from "../Dropdown";
+import type { NumberRange } from "../RangeInput";
+import { BlockNumberFilter } from "./BlockNumberFilter";
 import { RollupFilter } from "./RollupFilter";
 import { TimestampFilter } from "./TimestampFilter";
 
-const INIT_STATE = {
+const DEFAULT_MIN_VALUE = BigInt(0);
+const DEFAULT_MAX_VALUE = BigInt(99999999);
+
+const INIT_STATE: FiltersState = {
   rollup: null,
   timestampRange: {
     startDate: null,
     endDate: null,
   },
+  blockNumberRange: [undefined, undefined],
 };
 
+interface FiltersErrorsState {
+  blockNumberRange?: string;
+}
+
 interface FiltersState {
   rollup: Option | null;
   timestampRange: DateValueType;
+  blockNumberRange: NumberRange;
 }
 
 export const Filters: FC = function () {
   const [formData, setFormData] = useState<FiltersState>(INIT_STATE);
+  const [errors, setErrors] = useState<FiltersErrorsState>({
+    blockNumberRange: undefined,
+  });
+
   const router = useRouter();
 
   const breakpoint = useBreakpoint();
-  const fullWidth = breakpoint === "sm" || breakpoint === "default";
+  const fullWidth =
+    breakpoint === "md" || breakpoint === "sm" || breakpoint === "default";
```

Comment:
Suggestion to improve readability: `const fullWidth = ["md", "sm", "default"].includes(breakpoint);`